src/project/project-actions.tsx

```diff
@@ -245,6 +248,35 @@ export class ProjectActions {
     ));
   }
 
+  loadMlJson = async (
+    files: File[],
+    _type: LoadType = "file-upload"
+  ): Promise<void> => {
+    if (files.length === 0) {
+      throw new Error("Expected to be called with at least one file");
+    }
+
+    // Avoid lingering messages related to the previous project.
+    // Also makes e2e testing easier.
+    this.actionFeedback.closeAll();
+
+    if (files.length === 1 && getLowercaseFileExtension(files[0].name)) {
+      const file = files[0];
+      const json = await readFileAsText(file);
+      const data = JSON.parse(json) as ActionData[];
+      this.setMlData(data);
+      const actionNames = data.map((action) => action.name).join(" ");
```

Comment:
Action names can have spaces in so this will break. Probably better to have a name per line, but this will require a MicroPython update to change.